services/src/main/java/org/jboss/windup/web/services/websocket/ExecutionProgressReporter.java

```diff
@@ -59,10 +65,39 @@ public void onClose(Session session, CloseReason reason, @PathParam("executionId
         }
     }
 
+    @OnOpen
+    public void open(Session session, EndpointConfig config)
+    {
+        // Get the thread local context (created by WebSocketContextConfigurator) and attach the
+        // data to the session
+
+        if (session instanceof UndertowSession)
+        {
+            UndertowSession undertowSession = (UndertowSession) session;
+            WebSocketChannel webSocketChannel = undertowSession.getWebSocketChannel();
+            String url = webSocketChannel.getUrl();
+            if (!StringUtils.isBlank(url) && url.startsWith("ws://"))
+            {
+                try
+                {
+                    String serverHost = url.substring(5);
+                    serverHost = serverHost.substring(0, serverHost.indexOf("/"));
+                    session.getUserProperties().put("Host", serverHost);
+                }
+                catch (Exception e)
+                {
+                    LOG.warning("Could not parse ws url (" + url + ") due to: " + e.getMessage());
+                }
+            }
+        }
+
+    }
+
     @OnMessage
     public void onMessage(Session session, String message, @PathParam("executionId") Long executionId)
     {
-        authenticate(session, message);
+        String serverHost = (String) session.getUserProperties().get("Host");
+        authenticate(serverHost, session, message);
```

Comment:
@jsight I think this is wrong. `serverHost` is host which sent websocket request. But it doesn't imply it is also keycloak host. When keycloak runs on different server, this won't work.